daemon/endpoint.go

```diff
@@ -160,21 +159,27 @@ func (d *Daemon) createEndpoint(epTemplate *models.EndpointChangeRequest, id str
 		return PutEndpointIDFailedCode, err
 	}
 
-	add := labels.NewLabelsFromModel(lbls)
+	addLabels := labels.NewLabelsFromModel(lbls)
 
-	if len(add) > 0 {
-		code, errLabelsAdd := d.updateEndpointLabels(id, add, labels.Labels{})
-		if errLabelsAdd != nil {
-			// XXX: Why should the endpoint remain in this case?
-			log.WithFields(logrus.Fields{
-				logfields.EndpointID:              id,
-				logfields.IdentityLabels:          logfields.Repr(add),
-				logfields.IdentityLabels + ".bad": errLabelsAdd,
-			}).Error("Could not add labels while creating an ep due to bad labels")
-			return code, errLabelsAdd
+	if len(addLabels) > 0 {
+		addLabels, _, ok := checkLabels(addLabels, nil)
+		if !ok {
+			return PutEndpointIDInvalidCode, fmt.Errorf("No valid label")
```

Comment:
This case can't be hit, check the implementation of `checkLabels()`.

Reply:
It can, if all the labels in `addLabels` are filtered out by the call to `FilterLabels` made in `checkLabels`.